### PR TITLE
[review] Klocwork fix PFW 3.1.0

### PR DESCRIPTION
**Pull request by tcahuzax**
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/323%23issuecomment-167816082%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/323%23issuecomment-167821501%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/323%23discussion_r48540847%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/323%23issuecomment-167816082%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22plz%20review%20%40krocard%20%22%2C%20%22created_at%22%3A%20%222015-12-29T15%3A45%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/11178583%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/tcahuzax%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%20%22%2C%20%22created_at%22%3A%20%222015-12-29T16%3A07%3A32Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20a03539a02b7560cf81a87d939253d5f6706e94fb%20test/test-subsystem/TESTSubsystemString.cpp%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/323%23discussion_r48540847%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20do%20not%20realy%20nead%20those%20%60std%3A%3A%60%20but%20it%20is%20more%20a%20matter%20of%20style.%22%2C%20%22created_at%22%3A%20%222015-12-29T13%3A44%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/test-subsystem/TESTSubsystemString.cpp%3AL49-55%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/krocard%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/krocard'></a>

- [ ] <a href='#crh-comment-Pull a03539a02b7560cf81a87d939253d5f6706e94fb test/test-subsystem/TESTSubsystemString.cpp 16'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/323#discussion_r48540847'>File: test/test-subsystem/TESTSubsystemString.cpp:L49-55</a></b>
- <a href='https://github.com/krocard'></a> You do not realy nead those `std::` but it is more a matter of style.
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/323#issuecomment-167816082'>General Comment</a></b>
- <a href='https://github.com/tcahuzax'></a> plz review @krocard

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/323?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/323?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/323?approve=0'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/323'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by dawagner** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/01org/parameter-framework/pull/323'>Refresh</a>.<a href='#crh-update-none'></a>

File: test/test-subsystem/TESTSubsystemString.cpp [hunk 1]
```diff
@@ -46,5 +49,6 @@ std::string CTESTSubsystemString::toString(const void *pvValue, size_t /*size*/)
 
 void CTESTSubsystemString::fromString(const std::string &strValue, void *pvValue, size_t size)
 {
-    strncpy((char *)pvValue, strValue.c_str(), size);
+    auto destination = MAKE_ARRAY_ITERATOR(static_cast<char *>(pvValue), size);
+    std::copy_n(std::begin(strValue), size, destination);
```
Comment (krocard) [inline test/test-subsystem/TESTSubsystemString.cpp]: You do not realy nead those `std::` but it is more a matter of style.

**Comment by tcahuzax** [comment]
plz review @krocard

**Comment by krocard** [comment]
:shipit:

**Pull request merged**